crypto/provider_core.c

```diff
@@ -2238,11 +2254,13 @@ static const OSSL_DISPATCH core_dispatch_[] = {
     { OSSL_FUNC_BIO_VSNPRINTF, (void (*)(void))BIO_vsnprintf },
     { OSSL_FUNC_SELF_TEST_CB, (void (*)(void))core_self_test_get_callback },
     { OSSL_FUNC_GET_ENTROPY, (void (*)(void))rand_get_entropy },
+    { OSSL_FUNC_GET_USER_ENTROPY, (void (*)(void))rand_get_user_entropy },
     { OSSL_FUNC_CLEANUP_ENTROPY, (void (*)(void))rand_cleanup_entropy },
+    { OSSL_FUNC_CLEANUP_USER_ENTROPY, (void (*)(void))rand_cleanup_user_entropy },
     { OSSL_FUNC_GET_NONCE, (void (*)(void))rand_get_nonce },
-    { OSSL_FUNC_CLEANUP_NONCE, (void (*)(void))rand_cleanup_nonce },
-    { OSSL_FUNC_GET_USER_ENTROPY, (void (*)(void))rand_get_user_entropy },
     { OSSL_FUNC_GET_USER_NONCE, (void (*)(void))rand_get_user_nonce },
+    { OSSL_FUNC_CLEANUP_NONCE, (void (*)(void))rand_cleanup_nonce },
+    { OSSL_FUNC_CLEANUP_USER_NONCE, (void (*)(void))rand_cleanup_user_nonce },
 #endif
```

Comment:
So you're adding these new core callbacks but I do not see them being used anywhere??

Reply:
Good catch! I seems like I overlooked to add them to `ossl_prov_seeding_from_dispatch`:

https://github.com/openssl/openssl/blob/143ca66cf00c88950d689a8aa0c89888052669f4/providers/common/provider_seeding.c#L48-L68

Reply:
I added the missing entries and split off the cleanup callbacks into a [separate commit](https://github.com/openssl/openssl/pull/22394/commits/ad112210e847f6b5b9adbcf4df6794b30939bdd8). Hope this makes the changes easier to review.